ReactiveCocoa/Swift/Signal.swift

```diff
@@ -680,11 +680,29 @@ public func tryMap<T, U, E>(operation: T -> Result<U, E>)(signal: Signal<T, E>)
 	}
 }
 
+/// Forwards events from `signal` until `interval`. Then if signal isn't completed yet,
+/// errors with `error` on `scheduler`.
+public func timeoutWithError<T, E>(error: E, afterInterval interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType)(signal: Signal<T, E>) -> Signal<T, E> {
+	precondition(interval >= 0)
```

Comment:
> What if we documented it instead?

I was going to suggest this too. I got a bit carried away with that sentence about this being suspicious.